AsyncUsageAnalyzers/AsyncUsageAnalyzers.Test/Reliability/AvoidAsyncVoidUnitTests.cs

```diff
@@ -0,0 +1,108 @@
+﻿namespace AsyncUsageAnalyzers.Test.Reliability
+{
+    using System.Threading;
+    using System.Threading.Tasks;
+    using AsyncUsageAnalyzers.Reliability;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using TestHelper;
+    using Xunit;
+
+    public class AvoidAsyncVoidUnitTests : DiagnosticVerifier
+    {
+        [Fact]
+        public async Task TestAsyncReturnVoidAsync()
+        {
+            string testCode = @"
+class ClassName
+{
+    async void MethodNameAsync() { }
+}
+";
+
+            DiagnosticResult expected = CSharpDiagnostic().WithArguments("MethodNameAsync").WithLocation(4, 16);
+            await VerifyCSharpDiagnosticAsync(testCode, expected, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestAsyncEventHandlerReturnVoidAsync()
+        {
+            string testCode = @"
+using System;
+class ClassName
+{
+    ClassName()
+    {
+        AppDomain.CurrentDomain.DomainUnload += MethodNameAsync;
+    }
+
+    async void MethodNameAsync(object sender, EventArgs e) { }
+}
+";
+
+            // This analyzer does not currently handle this case differently from any other method
+            DiagnosticResult expected = CSharpDiagnostic().WithArguments("MethodNameAsync").WithLocation(10, 16);
+            await VerifyCSharpDiagnosticAsync(testCode, expected, CancellationToken.None).ConfigureAwait(false);
+        }
```

Comment:
I think it would be better to write to test case to cover the intended situation and then skip the fact with a remark that the analyzer does not currently handle it properly.